File: frontend/src/model/Image.model.ts
import { Dayjs } from 'dayjs';

import { IImageLink } from '@/model/ImageLink.model';

export interface IImage {
    id: string;
    title: string | null;
    description: string | null;
    createdAt: Dayjs;
    width: number;
    height: number;
    fileSize: number;
    deleteHash: string;
    image: IImageLink;
}

File: backend/Models/ImageModel.cs
namespace Models;

public sealed class ImageModel
{
    public string Id { get; init; } = null!;
    public string? Title { get; init; }
    public string? Description { get; init; }
    public DateTime CreatedAt { get; init; }
    public int Width { get; init; }
    public int Height { get; init; }
    public int FileSize { get; init; }
    public string DeleteHash { get; init; } = null!;
}